### PR TITLE
Quark W4A8 SVDQuant Support

### PR DESCRIPTION
**Pull request by amd-xiaoyu12**
## Purpose
Adds **W4A8** quantization for Wan2.2 diffusion transformers on ROCm/gfx950 (MI355X): MXFP4 weights + dynamically-quantized MXFP8 activat>

**Three accuracy tiers, two `--quantization` methods** (all one kernel launch):

| Tier | `--quantization` | Correction | Offline setup |
| --- | --- | --- | --- |
| plain W4A8 (RTN) | `quark_w4a8` | none | none |
| online W4A8 SVD | `quark_svdquant` | low-rank, derived at load (uncalibrated) | none |
| calibrated W4A8 SVD | `quark_svdquant` | low-rank, activation-calibrated | one export |

The two SVD tiers are the same method; the checkpoint (`is_checkpoint_w4a8_serialized`) selects online vs calibrated.

- **Online** (stock BF16): plain RTN; SVD derives `L1`/`L2` with `torch.svd_lowrank` (an accuracy floor  ^`^t uncalibrated).
- **Offline** (`examples/quantization/export_quark_svdquant_w4a8.py`): runs Quark's calibrated `SVDQuantProcessor` (SmoothQuant + exact S>
- **On-disk formats** (`--pack-format`): `bf16` (default), `mxfp4_packed` (preshuffled, TP=1, fastest load), `mxfp4_unshuffled` (natural->
- **Tensor parallelism** on `mxfp4_unshuffled`: plain and SVD both support column- and row-parallel. Row-parallel SVD needs **no extra co>

gfx950-only, loud BF16 fallback for untileable layers. Quark is imported lazily, so the serve path stays Quark-free.

> **Requires** AMD Quark PR [#6079](https://gitenterprise.xilinx.com/AMDNeuralOpt/Quark/pull/6079) (branch `xiaoyu/svd-quant-flydsl`)

## Test Plan

**vLLM Version:**

**vLLM-Omni Commit:**

## Test Result

**BEFORE SUBMITTING:** read [CONTRIBUTING.md](https://github.com/vllm-project/vllm-omni/blob/main/CONTRIBUTING.md) and run the [precheck-pr skill](https://github.com/vllm-project/vllm-omni/blob/main/.claude/skills/precheck-pr/SKILL.md) with the code agent for a self-check against project conventions.
(anything written below this line will be removed by GitHub Actions)

### CODE REVIEW

File: vllm_omni/quantization/quark_w4a8_config.py [hunk 1]
```diff
@@ -0,0 +1,650 @@
...
+        self,
+        layer: torch.nn.Module,
+        input_size_per_partition: int,
+        output_partition_sizes: list[int],
+        input_size: int,
+        output_size: int,
+        params_dtype: torch.dtype,
+        **extra_weight_attrs,
+    ) -> None:
+        self._check_parallelism(layer, input_size_per_partition, input_size)
```
Comment (lishunyang12) [inline vllm_omni/quantization/quark_w4a8_config.py:479]: Check per-rank K/N here; A14B TP4 admits unsupported 3456-wide shards.

File: docs/user_guide/quantization/w4a8_rocm.md [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+
+where `<comp>` is `transformer` (plus `transformer_2` for the A14B cascade).
+Weights are stored **unpacked BF16** (residual under `weight`, factors under
+`proj_down`/`proj_up`) and packed to the MXFP4 kernel layout at load. Copy or
+symlink the non-transformer pipeline components (`vae`, `text_encoder`,
+`scheduler`, `model_index.json`) into `<output_dir>` to make it directly
+loadable, then serve it exactly like a stock model:
+
+```bash
+vllm serve /path/to/wan5b-w4a8-svd --omni --quantization quark_w4a8
```
Comment (lishunyang12) [inline docs/user_guide/quantization/w4a8_rocm.md:165]: Drop --quantization here so the checkpoint's serialized SVD configuration is preserved.

File: examples/quantization/export_quark_svdquant_w4a8.py [hunk 3]
```diff
@@ -0,0 +1,403 @@
...
+        write_component(args.output_dir, comp, sd, args)
+
+    # Symlink the non-transformer pipeline components so --output_dir is directly
+    # loadable (the transformer dirs are the quantized exports we just wrote).
+    for entry in os.listdir(args.model):
+        if entry in comps:
+            continue
+        dst = os.path.join(args.output_dir, entry)
+        if not os.path.exists(dst):
+            os.symlink(os.path.join(args.model, entry), dst)
```
Comment (lishunyang12) [inline examples/quantization/export_quark_svdquant_w4a8.py:396]: Resolve args.model to an absolute path before creating these symlinks.
